docs/data/tree-view/tree-item-customization/tree-item-customization.md

```diff
@@ -146,13 +146,20 @@ You can use the `useTreeItem` hook to access the `publicAPI` object from within
 
 See the **Imperative API** section on each feature page to learn more about the public API methods available on the Tree View.
 
+:::warning
+The `publicAPI` object should not be used in the render because the item won't necessary re-render when the value returned is updated.
```

Comment:
```suggestion
The `publicAPI` object should not be used in the render because the item won't necessarily re-render when the returned value is updated.
```